projects/igniteui-angular/src/lib/grid/grid.component.ts

```diff
@@ -234,9 +234,9 @@ export class IgxGridComponent implements OnInit, OnDestroy, AfterContentInit, Af
 	 * @memberof IgxGridComponent
 	 */
     set filteringExpressionsTree(value) {
-        if (value) {
-            for (let index = 0; index < (value as FilteringExpressionsTree).filteringOperands.length; index++) {
-                const val = (value as FilteringExpressionsTree);
+        if (value && value instanceof FilteringExpressionsTree) {
```

Comment:
The good practice is to check if a value is instance of an interface (if any). In this case this should be `value instance of IFilteringExpressionsTree`.